src/solvers/smt2_incremental/smt_sorts.cpp

```diff
@@ -10,6 +10,16 @@
 #include <solvers/smt2_incremental/smt_sorts.def>
 #undef SORT_ID
 
+bool smt_sortt::operator==(const smt_sortt &other) const
+{
+  return irept::operator==(other);
```

Comment:
Question: since this just calls the base class' method, with not other change to it, does it make sense to kill this method here and just `using irept::operator==;` in the header file?

Reply:
The `irept::operator==` member function takes an `irept` as its parameter / right hand side. So yes we could add `using irept::operator==;`, but the existing tests which use the `smt_sortt::operator==` would not compile because the implicit upcast to `irept` is banned from outside of the data structure's member functions due to the use of protected inheritance. Also I don't think it would make sense to allow comparisons such as `smt_bool_sortt{} == irept{}`. 1) because sorts should not be stored as `irept` outside of the irep based data structures and 2) because if we know that one side is not a sort at compile time and that the other side is a sort, then we know that the result would always be false.